src/test/java/seedu/savenus/model/purchase/PurchaseHistoryTest.java

```diff
@@ -11,16 +13,14 @@
 import org.junit.jupiter.api.Test;
 
 import javafx.collections.FXCollections;
-import seedu.savenus.model.food.Name;
-import seedu.savenus.model.food.Price;
 import seedu.savenus.model.purchase.exceptions.PurchaseNotFoundException;
 
 public class PurchaseHistoryTest {
-    private final Purchase testPurchase = new Purchase(new Name("testFoodName"), new Price("1.50"));
+    private final Purchase testPurchase = new Purchase(CARBONARA);
     private final PurchaseHistory testPurchaseHistory = new PurchaseHistory();
     private final List<Purchase> testListOfPurchases = new ArrayList<>(List.of(
```

Comment:
Maybe you could consider using a HashSet like Jon. If food item has already been added, then you dont have to save this food item again. Then save the timings in a separate data structure and add the references to these timings in the Food object? 

However, at this point, this works perfectly fine!!! :)